libraries/resource_providers/api500/c7000/volume_provider.rb

```diff
@@ -57,14 +53,14 @@ def create_from_snapshot
           end
         end
 
-        def add_or_edit
-          validate_required_properties(:storage_system, :device_volume_name, :is_shareable)
-          options = Marshal.load(Marshal.dump(@item.data))
-          return Chef::Log.info "#{@resource_name} '#{@name}' already exists" if @item.exists?
-          storage_system = resource_named(:StorageSystem).new(@item.client, name: @new_resource.storage_system)
+        def add_if_missing
```

Comment:
I think add_if_missing more coherent, since editing can be done through create action. If you don't agree, I will change to the previous form.